force-app/main/default/classes/DegreeTriggerHelper.cls

```diff
@@ -0,0 +1,124 @@
+public with sharing class DegreeTriggerHelper {
+  //  public static void preventUnrelatedClassSubject(List<ClassEnrollment__c> enrollmentBeingInserted) {
+  //      checkIfClassSubjectMatchesStudentMajor(enrollmentBeingInserted);
+  //  }
+
+    public static void preventUnrelatedClassSubject(ClassEnrollment__c enrollmentBeingInserted) {
+        checkIfClassSubjectMatchesStudentMajor(enrollmentBeingInserted);
+    }
+
+    static void checkIfClassSubjectMatchesStudentMajor(ClassEnrollment__c newEnrollment){
+        Id studentId = newEnrollment.Student__c;
+        Id classId = newEnrollment.Class__c;
+
+        Class__c classDept = [SELECT Department__r.Name FROM Class__c WHERE Id =: classId];
+        Student__c studentDept = [SELECT (SELECT Degree__r.Department__r.Name FROM Majors__r) FROM Student__c WHERE Id =: studentId];
+
+        if (studentDept.get('Name') != classDept.get('Name')){
```

Comment:
Checking Id would be better here since Name is not always guaranteed to be unique.